lib/settings-observer.js

```diff
@@ -52,11 +52,15 @@ class SettingsObserver {
   get handler() {
     return this._handler;
   }
-  populate() {
-    const newVal = _.get(this._settings._cacheObj, this._keyPath);
-    if (!_.isEqual(newVal, this._currentValue)) {
-      this._handler.call(this, newVal, this._currentValue);
-      this._currentValue = newVal;
+  onChange() {
+    try {
+      const newVal = _.get(this._settings._cacheObj, this._keyPath);
+      if (!_.isEqual(newVal, this._currentValue)) {
+        this._handler.call(this, newVal, this._currentValue);
+        this._currentValue = newVal;
```

Comment:
This should be before the handler call? To make sure we store the new value in case handler exception

Reply:
We don't really need a copy of oldVal?